base/osutils.jl

```diff
@@ -1,62 +1,14 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
 """
-    is_unix([os])
-
-Predicate for testing if the OS provides a Unix-like interface.
-See documentation in [Handling Operating System Variation](@ref).
-"""
-function is_unix(os::Symbol)
-    if is_windows(os)
-        return false
-    elseif is_linux(os) || is_bsd(os)
-        return true
-    else
-        throw(ArgumentError("unknown operating system \"$os\""))
-    end
-end
-
-"""
-    is_linux([os])
-
-Predicate for testing if the OS is a derivative of Linux.
-See documentation in [Handling Operating System Variation](@ref).
-"""
-is_linux(os::Symbol) = (os == :Linux)
-
-"""
-    is_bsd([os])
-
-Predicate for testing if the OS is a derivative of BSD.
-See documentation in [Handling Operating System Variation](@ref).
-"""
-is_bsd(os::Symbol) = (os == :FreeBSD || os == :OpenBSD || os == :NetBSD || os == :DragonFly || os == :Darwin || os == :Apple)
-
-"""
-    is_windows([os])
-
-Predicate for testing if the OS is a derivative of Microsoft Windows NT.
-See documentation in [Handling Operating System Variation](@ref).
-"""
-is_windows(os::Symbol) = (os == :Windows || os == :NT)
-
-"""
-    is_apple([os])
-
-Predicate for testing if the OS is a derivative of Apple Macintosh OS X or Darwin.
-See documentation in [Handling Operating System Variation](@ref).
-"""
-is_apple(os::Symbol) = (os == :Apple || os == :Darwin)
-
-"""
     @static
 
 Partially evaluates an expression at parse time.
 
-For example, `@static is_windows() ? foo : bar` will evaluate `is_windows()` and insert either `foo` or `bar` into the expression.
+For example, `@static iswindows() ? foo : bar` will evaluate `iswindows()` and insert either `foo` or `bar` into the expression.
 This is useful in cases where a construct would be invalid on other platforms,
 such as a `ccall` to a non-existent function.
-`@static if is_apple() foo end` and `@static foo <&&,||> bar` are also valid syntax.
+`@static if isapple() foo end` and `@static foo <&&,||> bar` are also valid syntax.
 """
 macro static(ex)
```

Comment:
could perhaps move this to another file and delete osutils.jl then, but doesn't have to be part of this pr